airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -105,6 +105,22 @@ def test_read(
 
         assert records, "At least one record should be read using provided catalog"
 
+        for stream in [stream for stream in configured_catalog.streams if stream.stream.source_defined_primary_key]:
+            stream_records = [r for r in records if r.stream == stream.stream.name]
+            for stream_record in stream_records:
+                for sub_pk in stream.stream.source_defined_primary_key:
+                    record_data = stream_record.data
+                    for sub_pk_part in sub_pk:
+                        if isinstance(record_data, dict):
+                            record_data = record_data.get(sub_pk_part)
+                        else:
+                            record_data = None
+
+                        assert record_data, (
+                            f"Primary key subkeys {repr(sub_pk)} "
+                            f"have null values or not present in {stream.stream.name} stream records. "
```

Comment:
There is an odd space here at the end of the line before `"`.

Reply:
@htrueman is it necessary to show message that certain primary key missed? Or will be enough message, that some PK is missed?

Reply:
I think showing message that certain primary key missed is much more convenient, so I've added this log on purpose.
If we have composed PK like this: `[["pk1", "pk2"], ["pk3", "pk4"]]`, then this log will show you which exact part of the PK is broken so you could debug it closely.